src/format.js

```diff
@@ -104,11 +104,26 @@ function extractMetrics (trace, span) {
   }
 }
 
-function addTag (meta, key, value) {
-  value = serialize(value)
-
-  if (typeof value === 'string') {
-    meta[key] = value
+function addTag (meta, key, value, depth) {
+  depth = depth || 0
+
+  switch (typeof value) {
+    case 'string':
+      meta[key] = value
+      break
+    case 'undefined':
+      break
+    case 'object':
+      if (value === null) break
```

Comment:
Should we unset in this case?

Reply:
Good point. I wanted to do it but forgot. Should the same apply to `undefined`?

Reply:
I originally was going to leave a comment there too, but thinking:

```js
// Normal
span.setTag('custom.tag', true);

// Undefined
span.setTag('custom.tag');
span.setTag('custom.tag', undefined);

// Null
span.setTag('custom.tag', null);
```

undefined is a weird one, since it sort of depends, if they intentionally mean to set to `undefined` we don't know because we see it as the same as the first one. And setting with no value feels like it should be handled than an explicit value... so I am thinking we just ignore it... ?

Reply:
We could know whether it was explicitly passed or not, but I'm not sure that we should have 2 different behaviors for `undefined`. In general if they want to set `undefined` they should probably pass it as a string since it will be converted to a string anyway.

I'm fine with it either way, but I myself has been bit a few times by having `undefined` appear in the UI in a few unexpected cases. `env: undefined` is still haunting my account to this day, so my preference is to unset for both.

Reply:
I just realized that the behavior is already to unset both `null` and `undefined` with the current code since it's handled beforehand by the `Span` class.